x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -239,47 +240,23 @@ public static ElementType toElementType(DataType dataType) {
      * For example, spatial types can be extracted into doc-values under specific conditions, otherwise they extract as BytesRef.
      */
     public static ElementType toElementType(DataType dataType, MappedFieldType.FieldExtractPreference fieldExtractPreference) {
-        if (dataType == DataType.LONG
-            || dataType == DataType.DATETIME
-            || dataType == DataType.UNSIGNED_LONG
-            || dataType == DataType.COUNTER_LONG) {
-            return ElementType.LONG;
-        }
-        if (dataType == DataType.INTEGER || dataType == DataType.COUNTER_INTEGER) {
-            return ElementType.INT;
-        }
-        if (dataType == DataType.DOUBLE || dataType == DataType.COUNTER_DOUBLE) {
-            return ElementType.DOUBLE;
-        }
-        // unsupported fields are passed through as a BytesRef
-        if (dataType == DataType.KEYWORD
-            || dataType == DataType.TEXT
-            || dataType == DataType.IP
-            || dataType == DataType.SOURCE
-            || dataType == DataType.VERSION
-            || dataType == DataType.UNSUPPORTED) {
-            return ElementType.BYTES_REF;
-        }
-        if (dataType == DataType.NULL) {
-            return ElementType.NULL;
-        }
-        if (dataType == DataType.BOOLEAN) {
-            return ElementType.BOOLEAN;
-        }
-        if (dataType == DataType.DOC_DATA_TYPE) {
-            return ElementType.DOC;
-        }
-        if (dataType == DataType.TSID_DATA_TYPE) {
-            return ElementType.BYTES_REF;
-        }
-        if (EsqlDataTypes.isSpatialPoint(dataType)) {
-            return fieldExtractPreference == DOC_VALUES ? ElementType.LONG : ElementType.BYTES_REF;
-        }
-        if (EsqlDataTypes.isSpatial(dataType)) {
+
+        return switch (dataType) {
+            case LONG, DATETIME, UNSIGNED_LONG, COUNTER_LONG -> ElementType.LONG;
+            case INTEGER, COUNTER_INTEGER -> ElementType.INT;
+            case DOUBLE, COUNTER_DOUBLE -> ElementType.DOUBLE;
+            // unsupported fields are passed through as a BytesRef
+            case KEYWORD, TEXT, IP, SOURCE, VERSION, UNSUPPORTED -> ElementType.BYTES_REF;
+            case NULL -> ElementType.NULL;
+            case BOOLEAN -> ElementType.BOOLEAN;
+            case DOC_DATA_TYPE -> ElementType.DOC;
+            case TSID_DATA_TYPE -> ElementType.BYTES_REF;
+            case GEO_POINT, CARTESIAN_POINT -> fieldExtractPreference == DOC_VALUES ? ElementType.LONG : ElementType.BYTES_REF;
             // TODO: support forStats for shape aggregations, like st_centroid
```

Comment:
This comment is out of date, and can be removed. When we do doc-values they will almost certainly also be backed by ByteRefs, so no consideration for fieldExtractPreference is needed (the term `forStats` was used before we moved to FieldExtractPreference).